src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -734,6 +735,29 @@ internal static ImmutableArray<T> Concat<T>(this ImmutableArray<T> first, T seco
             return first.Add(second);
         }
 
+        internal static ImmutableArray<T> AddRange<T>(this ImmutableArray<T> self, in TemporaryArray<T> items)
+        {
+            if (items.Count == 0)
+            {
+                return self;
+            }
+
+            if (items.Count == 1)
+            {
+                return self.Add(items[0]);
+            }
+
+            var builder = ArrayBuilder<T>.GetInstance(self.Length + items.Count);
+            builder.AddRange(self);
+
+            foreach (var item in items)
+            {
+                builder.Add(item);
+            }
```

Comment:
```suggestion
            builder.AddRange(items);
```

Reply:
The builder does not have an overload that takes `TemporaryArray<T>`